### PR TITLE
In feed engagement modules

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

**Comment by robbiecarlton** [comment]
attn @lorenjohnson - this now depends on 
https://github.com/Hylozoic/hylo-node/pull/202
so you should switch to that branch in node before working on this.

**Comment by robbiecarlton** [comment]
screen shot 2017-01-03 at 3 08 29 pm

**Comment by robbiecarlton** [comment]
screen shot 2017-01-03 at 4 01 52 pm
screen shot 2017-01-03 at 4 02 02 pm

**Comment by levity** [comment]
n.b. i changed the sendGraphqlQuery action to handle parameters properly: https://github.com/Hylozoic/hylo-redux/commit/ab02187bacb2732561c4a6ced46ce124128ae31c

**Comment by levity** [comment]
hmm, merge conflict. can you resolve it by moving both `variables` and `addDataToStore` into an options object so we don't have too many unnamed arguments?

**Comment by lorenjohnson** [comment]
Build failing on CircleCI with a HyloUtils related issue: "TypeError: (0 , _hyloUtils.groupUser) is not a function"

I am not getting this locally nor do I know how to fix it. However, locally I did get this same issue regardless of removing HyloUtils and reinstalling with yarn. I was able to fix it locally by running both a yarn install and an npm install :-/

(accidentally closed the PR, then re-opened. sorry)

**Comment by levity** [comment]
going to hold off on this until #225 is resolved -- i have the same concerns about completing the refactor of `sendGraphqlQuery` and the use of `sendGraphqlQueryAddDataToStore` that i described there.

File: src/components/PopularSkillsModule.js
Comment (robbiecarlton) [inline src/components/PopularSkillsModule.js:13]: @levity compare this function call with the call here:
https://github.com/Hylozoic/hylo-redux/blob/d39666a4d9865e6ff1f62ef83dd51bd8ec4e5255/src/components/PopularSkillsModule.js
the extra nesting, and the extra 'data' component of all the `get` paths (which obfuscates the relationship between the query structure and the payload structure) are why I wanted to have a separate `sendGraphqlQueryAddDataToStore` function.

I wonder if, seeing this, you still want to just use the basic `sendGraphqlQuery` here? And if so, I'd be curious to hear you're reasoning.

Comment (levity) [reply src/components/PopularSkillsModule.js:13]: This reads more clearly to me because the `addDataToStore` key is associated with the relevant contents rather than the whole function. We can remove the repetitive use of `data` by post-processing the response from the API with `meta.then`:

```js
return {
  type: SEND_GRAPHQL_QUERY,
  payload: {...},
  meta: {
    subject,
    id,
    then: get('data'),
    addDataToStore,
  }
}
```

Which reminds me that we have to handle GraphQL errors properly. But I can do that later, once this is merged.

File: src/components/PopularSkillsModule.js
Comment (levity) [inline src/components/PopularSkillsModule.js]: you can also omit `id` here

File: src/containers/ConnectedPostList.js
Comment (levity) [inline src/containers/ConnectedPostList.js]: you can still declare this `const` if you're just mutating it, not re-assigning

File: src/containers/People.js
Comment (levity) [inline src/containers/People.js:97]: why remove this?

File: src/containers/community/CommunityPosts.js
Comment (levity) [inline src/containers/community/CommunityPosts.js]: this can be a top-level function that just takes community and moduleChoice and returns module

File: src/util/index.js
Comment (levity) [inline src/util/index.js]: looks like this isn't used? if you want to keep it in for debugging purposes, let's call it tapLog instead

File: test/client/components/CommunityPosts.test.js
Comment (levity) [inline test/client/components/CommunityPosts.test.js]: if it doesn't refer to an actual object let's use normal words instead of camelcase

Comment (levity) [reply test/client/components/CommunityPosts.test.js]: nice use of mocks here

File: test/components/PostList.test.js
Comment (levity) [inline test/components/PostList.test.js]: is this comment still relevant?

File: src/containers/community/CommunityPosts.js
Comment (levity) [inline src/containers/community/CommunityPosts.js]: i think this would read a bit better if you did 
```js
if (showPopularSkills && showPostPrompt)
else if (showPopularSkills)
else if (showPostPrompt)
else
```

File: src/containers/People.js
Comment (robbiecarlton) [reply src/containers/People.js:97]: not sure. By mistake. Putting it back.

**Comment by robbiecarlton** [comment]
@lorenjohnson, Lawrence's remaining comments and the merge conflict are about the tests, so I'm going to leave them for you.

**Comment by lorenjohnson** [comment]
K, I'll look at it

**Comment by lorenjohnson** [comment]
Agree with all remaining suggestions and have remedied the missing test cast for PostList modules. I think this should be ready to clear out of the gate now.

**Review by levity — APPROVED** [review]
:shipit:

**Comment by levity** [comment]
hey, Github's online merge conflict editor is pretty sweet.

**Comment by levity** [comment]
Will let you guys handle merging and setting up feature flags.

**Pull request merged**
